### DISCUSSION TITLE
Guidelines on porting Arduino_Core_STM32 on MCU

### DISCUSSION THREAD

**Discussion by geocheats2** [stm32duino/Arduino_Core_STM32#1469] [Q&A]
Hello to everyone could someone explain the basic steps i need to take in order to port this for a custom pcb i designed based on 

[STM32F407IGT6](https://datasheet.lcsc.com/szlcsc/1809200016_STMicroelectronics-STM32F407IGT6_C33993.pdf)

This one is pretty much the same with its smaller "brothers" with the exception of more pins available
(ok there is more but lets just start somewhere)

Keep in mind, just the procedure required, i don't ask for a port

Looking forward to your replies
George

[Nevermind i am blind](https://github.com/stm32duino/wiki/wiki/Add-a-new-variant-(board))

**Comment by fpistm** [comment]
> Nevermind i am blind

😉

**Reply by fpistm** [reply]
I know I do not end the Wiki, anyway you have an example in the PR used as example in the Wiki, mainly this commit:
https://github.com/stm32duino/Arduino_Core_STM32/pull/1398/commits/20c7989a41d7588e397c5d7496046c118f9c4264

**Comment by geocheats2** [comment]
Hey i am back :) 
I get those errors trying to compile blink

arm-none-eabi-g++: error: {build.enable_usb}: No such file or directory
arm-none-eabi-g++: error: {build.bootloader_flags}: No such file or directory
arm-none-eabi-g++: error: {build.vect_flags}: No such file or directory

While the [example](https://github.com/stm32duino/Arduino_Core_STM32/commit/20c7989a41d7588e397c5d7496046c118f9c4264) has the build flags set 

Nucleo_64.menu.pnum.NUCLEO_G0B1RE.build.extra_flags=-D{build.product_line} {build.enable_usb} {build.xSerial} -D__CORTEX_SC=0

The boards.txt of boards_entry.txt doesn't mention anything 

GenF4.menu.pnum.GENERIC_F407IGTX=Generic F407IGTx
GenF4.menu.pnum.GENERIC_F407IGTX.upload.maximum_size=1048576
GenF4.menu.pnum.GENERIC_F407IGTX.upload.maximum_data_size=131072
GenF4.menu.pnum.GENERIC_F407IGTX.build.board=GENERIC_F407IGTX
GenF4.menu.pnum.GENERIC_F407IGTX.build.product_line=STM32F407xx
GenF4.menu.pnum.GENERIC_F407IGTX.build.variant=STM32F4xx/F407I(E-G)(H-T)_F417I(E-G)(H-T)

I tried added them manualy

GenF4.menu.pnum.GENERIC_F407IGTX.build.extra.flags=-D{build.enable_usb} {build.bootloader_flags} {build.vect_flags}

But i still get the same error ?

I tried disabling all the options i could on arduino IDE but no success

**Reply by fpistm** [reply]
it seems there is something weird in the command line so it consider the {build_*} as a file instead of an option. 
Hard to tell what based on your input.

**Reply by fpistm** [reply]
Here an example of an F4 addition: https://github.com/stm32duino/Arduino_Core_STM32/commit/9207aaa3b29faa8133e62ebc49452a8e75324f31
